### PR TITLE
[kuberay] Add a test of the Ray Job Submission API to the KubeRay e2e tests.

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This PR modifies the KubeRay e2e autoscaling test so that one of its scaling commands is sent via the Ray Job Submission API.

This validates that the Ray Job Submission API works with KubeRay and, in particular, that the Ray Dashboard is correctly exposed.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] E2e test

### CODE REVIEW

File: python/ray/tests/kuberay/test_autoscaling_e2e.py [hunk 1]
```diff
@@ -325,12 +331,12 @@ def testAutoscaling(self):
         # Submit two {"Custom2": 3} bundles to upscale two workers with 5
         # Custom2 capacity each.
         logger.info("Scaling up workers with request for custom resources.")
-        kubectl_exec_python_script(
+        job_logs = ray_job_submit(
```
Comment (Jeffwan) [inline python/ray/tests/kuberay/test_autoscaling_e2e.py:334]: I assume we want to test both modes? Here we change from exec + client -> job submission instead?

File: python/ray/tests/kuberay/scripts/scale_up_custom.py [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
 
 
 def main():
-    """Submits custom resource request."""
+    """Submits custom resource request.
+
+    Also, validates runtime env data submitted with the Ray Job that executes
+    this script.
+    """
+    assert pytest.__version__ == "6.0.0"
```
Comment (Jeffwan) [inline python/ray/tests/kuberay/scripts/scale_up_custom.py:14]: This is used to make sure it matches with job submission dependencies? or it is the code dependency?

File: python/ray/tests/kuberay/utils.py [hunk 3]
```diff
@@ -334,3 +338,62 @@ def ray_client_port_forward(
...
+        for trie in range(1, 7):
+            time.sleep(5)
+            try:
+                client = JobSubmissionClient(f"http://127.0.0.1:{local_port}")
+            except ConnectionError as e:
+                if trie < 6:
+                    logger.info("Job client connection failed. Retrying in 5 seconds.")
+                else:
+                    raise e from None
+        job_id = client.submit_job(
```
Comment (Jeffwan) [inline python/ray/tests/kuberay/utils.py:378]: job submission part looks good to me

**Review by Jeffwan — APPROVED** [review]

File: python/ray/tests/kuberay/scripts/scale_up_custom.py (hunk 2 above)
Comment (DmitriGekhtman) [reply python/ray/tests/kuberay/scripts/scale_up_custom.py:14]: > make sure it matches with job submission dependencies

Yep. Will add a comment.

File: python/ray/tests/kuberay/test_autoscaling_e2e.py (hunk 1 above)
Comment (DmitriGekhtman) [reply python/ray/tests/kuberay/test_autoscaling_e2e.py:334]: `testAutoscaling` now covers three modes of interaction --

kubectl exec (used in the CPU request part)
Ray client (used in the GPU request part)
Job submission (used in the custom resource request part)

File: python/ray/tests/kuberay/scripts/scale_up_custom.py (hunk 2 above)
Comment (DmitriGekhtman) [reply python/ray/tests/kuberay/scripts/scale_up_custom.py:14]: The test logic is maybe a bit unfocused.

File: python/ray/tests/kuberay/test_autoscaling_e2e.py (hunk 1 above)
Comment (DmitriGekhtman) [reply python/ray/tests/kuberay/test_autoscaling_e2e.py:334]: Added some comments in the code.

**Review by architkulkarni — APPROVED** [review]
Job submission part looks good!

**Pull request merged**